### PR TITLE
Bug/#1421 ability to set user access to set or read watchlist on the basis of profiles

### PR DESCRIPTION
**Pull request by Limraj**

### CODE REVIEW

**Review by radek2s — APPROVED** [review]

**Comment by Limraj** [comment]
Pullrequest will reopen after corrections have been made.

**Review by radek2s — APPROVED** [review]

File: src/org/scada_lts/mango/service/UsersProfileService.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    public Optional<UsersProfileVO> getProfileByUserId2(int userId) {
+        List<UsersProfileVO> profiles = usersProfileDAO.selectUserProfileByUserId(userId);
+        if(profiles.isEmpty())
+            return Optional.empty();
+        if(profiles.size() > 1) {
+            LOG.warn(LIST_SIZE_IS_GREATER_THAN_1);
+            return Optional.empty();
+        }
+        return Optional.ofNullable(profiles.get(0));
+    }
```
Comment (arturro96) [inline src/org/scada_lts/mango/service/UsersProfileService.java:44]: This method has no usages and is the same as method below. Also there are unused imports in this file.

**Review by arturro96 — APPROVED** [review]

**Comment by Limraj** [comment]
Closed until the problem is resolved

File: src/org/scada_lts/dao/ViewDAO.java [hunk 2]
```diff
@@ -182,6 +214,15 @@
 				+ COLUMN_NAME_MVU_VIEW_ID+"=? and "
 				+ COLUMN_NAME_MVU_USER_ID+"=?";
 
+	private static final String WATCHLIST_USERS_PROFILES_SELECT_BASE_ON_USERS_PROFILE_ID = ""
+			+ "select "
+			+ COLUMN_NAME_UP_VIEW_ID+ ", "
+			+ COLUMN_NAME_UP_PERMISSION + " "
+			+ "from "
+			+ "viewusersprofiles "
```
Comment (radek2s) [inline src/org/scada_lts/dao/ViewDAO.java:222]: viewusersprofiles can generate exceptions.
Try to use `viewUsersProfiles` literal.

**Review by arturro96 — APPROVED** [review]

**Review by radek2s — APPROVED** [review]

**Review by arturro96 — APPROVED** [review]

**Comment by Limraj** [comment]
The getUsersWithProfile method has been added and is included in the export. If we have a profile selected, it should set the profile id in a given user, if we have a profile none and some permissions for datasource / datapoint should also be included in the export.

**Review by arturro96 — APPROVED** [review]

**Review by radek2s — APPROVED** [review]

**Pull request merged**